File: GUIJpa/GUIJpa/GUIJpa/GUI1/GUI1/entity/ProductosPedidos.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GUI1.entity
{
    class ProductosPedidos
    {
       

        private int cantidad;

        private Double precio;

        private DateTime fecha;

        private int idProducto;
        
        private int idPedido;


        public int IdProducto { get => idProducto; set => idProducto = value; }
        public int IdPedido { get => idPedido; set => idPedido = value; }
        public int Cantidad { get => cantidad; set => cantidad = value; }
        public Double Precio { get => precio; set => precio = value; }
        public DateTime Fecha { get => fecha; set => fecha = value; }





    }
}

File: api-lyjstore/src/main/java/com/lyjstore/apilyjstore/model/ProductosPedidos.java
package com.lyjstore.apilyjstore.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "productos_pedidos")
public class ProductosPedidos implements Serializable {

    @EmbeddedId
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private ProductosPedidosPK productosPedidosPK;

    @Column(name = "id_producto",updatable = false,insertable = false)
    private Long idProducto;

    @Column(name = "id_pedido",updatable = false,insertable = false)
    private Long idPedido;

    private short cantidad;

    private Double precio;

    private LocalDate fecha;

    @JoinColumn(name = "id_producto",referencedColumnName = "id",insertable = false,updatable = false)
    @ManyToOne
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Producto producto;

    @JoinColumn(name = "id_pedido",referencedColumnName = "id",insertable = false,updatable = false)
    @ManyToOne
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Pedido pedido;


    public ProductosPedidos(ProductosPedidosPK productosPedidosPK, short cantidad, Double precio, LocalDate fecha) {
        this.productosPedidosPK = productosPedidosPK;
        this.cantidad = cantidad;
        this.precio = precio;
        this.fecha = fecha;
    }

    public ProductosPedidos() {
    }

}
